### PR TITLE
[mobile-client-build] Add support for adding optional env vars

### PR DESCRIPTION
**Pull request by JameelB**
## Description
Update the client build form to add support to additional environment variables that will be made available in the Jenkinsfile.

## Progress
- [x] Add section to the client build form to add environment variables
- [x] Make these environment variables to be available in the Jenkinsfile.

## Verification Steps

1. Provision Mobile CI/CD
2. Provision a mobile client
3. Create client build (Specify an environment variable)
4. Go to the build config's yaml file.
5. Ensure that the defined environment variable can be seen under `spec.strategy.jenkinsPipelineStrategy.env`
6. Create another client build (Do not specify any environment variable)
7. Ensure that no environment variable is added to `spec.strategy.jenkinsPipelineStrategy.env`

## Additional Notes
Related JIRA Ticket - https://issues.jboss.org/browse/AEROGEAR-2814

### CODE REVIEW

**Review by dimitraz — APPROVED** [review]
Verified locally, working as expected :+1:

File: app/scripts/controllers/createClientBuild.js [hunk 1]
```diff
@@ -131,6 +131,15 @@ angular.module('openshiftConsole')
         };
       }
 
+      if (clientConfig.envVars) {
+        _.forEach(clientConfig.envVars, function(envVar) {
+          if (envVar.name) {
+            var env = _.omit(envVar, ['$$hashKey']);
+            buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push(env);
+          }
+        })
```
Comment (sedroche) [inline app/scripts/controllers/createClientBuild.js]: Missing semi-colon

File: app/scripts/controllers/createClientBuild.js [hunk 2]
```diff
@@ -131,6 +131,15 @@ angular.module('openshiftConsole')
         };
       }
 
+      if (clientConfig.envVars) {
+        _.forEach(clientConfig.envVars, function(envVar) {
+          if (envVar.name) {
+            var env = _.omit(envVar, ['$$hashKey']);
```
Comment (sedroche) [inline app/scripts/controllers/createClientBuild.js]: We may remove this if [this property](https://github.com/openshift/origin-web-console/blob/5de5ca414129549aab483c7e9ab08a87cbf0e4fa/app/views/directives/edit-environment-variables.html#L17) removes the `$$hash` property

Comment (sedroche) [reply app/scripts/controllers/createClientBuild.js]: Ok. I think `angular.toJson` is probably best.

```.js
var envVars = _.map(clientConfig.envVars, angular.toJson);
```
?

Comment (JameelB) [reply app/scripts/controllers/createClientBuild.js]: sure, will change it to that. :+1:

File: app/scripts/controllers/createClientBuild.js [hunk 3]
```diff
@@ -131,6 +131,14 @@ angular.module('openshiftConsole')
         };
       }
 
+      if (clientConfig.envVars) {
+        _.forEach(clientConfig.envVars, function(envVar) {
+          if (envVar.name) {
```
Comment (JameelB) [inline app/scripts/controllers/createClientBuild.js:136]: @sedroche we still need to check for empty key value pairs as the YAML file won't accept this.

File: app/scripts/controllers/createClientBuild.js [hunk 4]
```diff
@@ -131,6 +131,14 @@ angular.module('openshiftConsole')
...
+            buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push(JSON.parse(angular.toJson(envVar)));
```
Comment (JameelB) [inline app/scripts/controllers/createClientBuild.js]: @sedroche Needed to parse after doing `angular.toJson` as this returns the object as a string.

**Review by sedroche — APPROVED** [review]

**Pull request merged**